h/static/scripts/directive/test/annotation-test.js

```diff
@@ -0,0 +1,1377 @@
+      }),
+      'private': sandbox.stub().returns({
+        read: ['justme']
+      }),
+      'default': sandbox.stub().returns({
+        read: ['default']
+      }),
+      setDefault: sandbox.stub()
+    };
+
+    fakePersonaFilter = sandbox.stub().returnsArg(0);
+
+    fakeDocumentTitleFilter = function(arg) {
+      return '';
+    };
+
+    fakeDocumentDomainFilter = function(arg) {
+      return '';
+    };
+
+    fakeSession = {
+      state: {
+        userid: 'acct:bill@localhost'
+      }
+    };
+
+    fakeTags = {
+      filter: sandbox.stub().returns('a while ago'),
+      store: sandbox.stub()
+    };
+
+    fakeTime = {
+      toFuzzyString: sandbox.stub().returns('a while ago'),
+      nextFuzzyUpdate: sandbox.stub().returns(30)
+    };
+
+    fakeUrlEncodeFilter = function(v) {
+      return encodeURIComponent(v);
+    };
+
+    fakeGroups = {
+      focused: function() {
+        return {};
+      },
+      get: function() {}
+    };
+
+    $provide.value('annotationMapper', fakeAnnotationMapper);
+    $provide.value('annotationUI', fakeAnnotationUI);
+    $provide.value('drafts', fakeDrafts);
+    $provide.value('features', fakeFeatures);
+    $provide.value('flash', fakeFlash);
+    $provide.value('momentFilter', fakeMomentFilter);
+    $provide.value('permissions', fakePermissions);
+    $provide.value('personaFilter', fakePersonaFilter);
+    $provide.value('documentTitleFilter', fakeDocumentTitleFilter);
+    $provide.value('documentDomainFilter', fakeDocumentDomainFilter);
+    $provide.value('session', fakeSession);
+    $provide.value('store', fakeStore);
+    $provide.value('tags', fakeTags);
+    $provide.value('time', fakeTime);
+    $provide.value('urlencodeFilter', fakeUrlEncodeFilter);
+    $provide.value('groups', fakeGroups);
+  }));
+
+  beforeEach(
+    inject(
+      function(_$compile_, _$document_, _$q_, _$rootScope_, _$timeout_,
+               _$window_) {
+        $compile = _$compile_;
+        $document = _$document_;
+        $window = _$window_;
+        $q = _$q_;
+        $timeout = _$timeout_;
+        $rootScope = _$rootScope_;
+        $scope = $rootScope.$new();
+        $scope.annotation = annotation = {
+          id: 'deadbeef',
+          document: {
+            title: 'A special document'
+          },
+          target: [{}],
+          uri: 'http://example.com',
+          user: 'acct:bill@localhost'
+        };
+      }
+    )
+  );
+
+  afterEach(function() {
+    sandbox.restore();
+  });
+
+  describe('when the annotation is a highlight', function() {
+    beforeEach(function() {
+      annotation.$highlight = true;
+      annotation.$create = sinon.stub().returns({
+        then: angular.noop,
+        'catch': angular.noop,
+        'finally': angular.noop
+      });
+    });
+
+    it('persists upon login', function() {
+      delete annotation.id;
+      delete annotation.user;
+      fakeSession.state.userid = null;
+      createDirective();
+      $scope.$digest();
```

Comment:
This `$scope.$digest()` call could be hoisted into `createDirective()`, if it isn't there already.